server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/VacationReply.java

```diff
@@ -0,0 +1,115 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.mailet;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.activation.DataHandler;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+import javax.mail.util.ByteArrayDataSource;
+
+import org.apache.mailet.Mail;
+import org.apache.mailet.MailAddress;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+
+public class VacationReply {
+
+    public static class Builder {
+
+        private final Mail originalMail;
+        private MailAddress mailRecipient;
+        private String reason;
+
+        public Builder(Mail originalMail) {
+            Preconditions.checkNotNull(originalMail, "Origin mail shall not be null");
+            this.originalMail = originalMail;
+        }
+
+        public Builder mailRecipient(MailAddress mailRecipient) {
+            this.mailRecipient = mailRecipient;
+            return this;
+        }
+
+        public Builder reason(String reason) {
+            this.reason = reason;
+            return this;
+        }
+
+        public VacationReply build() throws MessagingException {
+            Preconditions.checkNotNull(mailRecipient, "Original recipient address should not be null");
```

Comment:
please understand this : 
checkNotNull is for arguments
checkArgument is for arguments preconditions (but not for null checkà
checkState is for class state